spec/tunes/app_version_spec.rb

```diff
@@ -150,10 +171,308 @@
+          expect do
+            version.upload_screenshot!(screenshot_path, 0, "English", :iphone4)
+          end.to raise_error "sort_order must be positive"
+        end
+
+        it "prevents from using too large sort_order" do
+          expect do
+            version.upload_screenshot!(screenshot_path, 6, "English", :iphone4)
+          end.to raise_error "sort_order must not be > 5"
+        end
+
+        # not really sure if we want to enforce that
+        # it "prevents from letting holes in sort_orders" do
+        #  expect do
+        #    version.upload_screenshot!(screenshot_path, 4, "English", :iphone4)
+        #  end.to raise_error "FIXME"
+        # end
+
+        it "prevent from using invalid language" do
+          expect do
+            version.upload_screenshot!(screenshot_path, 1, "NotALanguage", :iphone4)
+          end.to raise_error "NotALanguage isn't an activated language"
+        end
+
+        it "prevent from using invalid language" do
+          expect do
+            version.upload_screenshot!(screenshot_path, 1, "English_CA", :iphone4)
+          end.to raise_error "English_CA isn't an activated language"
+        end
+
+        it "prevent from using invalid device" do
+          expect do
+            version.upload_screenshot!(screenshot_path, 1, "English", :android)
+          end.to raise_error "android isn't a valid device name"
+        end
+      end
+
+      describe "Add, Replace, Remove screenshots" do
+        it "can add a new screenshot to the list" do
+          du_upload_screenshot_success
+
+          count = version.screenshots["English"].count
+          version.upload_screenshot!(screenshot_path, 3, "English", :iphone4)
+          expect(version.screenshots["English"].count).to eq(count + 1)
+        end
+
+        it "can replace an existing screenshot with existing sort_order" do
+          du_upload_screenshot_success
+
+          count = version.screenshots["English"].count
+          version.upload_screenshot!(screenshot_path, 2, "English", :iphone4)
+          expect(version.screenshots["English"].count).to eq(count)
+        end
+
+        it "can remove existing screenshot" do
+          count = version.screenshots["English"].count
+          version.upload_screenshot!(nil, 2, "English", :iphone4)
+          expect(version.screenshots["English"].count).to eq(count - 1)
+        end
+
+        it "fails with error if the screenshot to remove doesn't exist" do
+          expect do
+            version.upload_screenshot!(nil, 3, "English", :iphone4)
+          end.to raise_error "cannot remove screenshot with non existing sort_order"
+        end
+      end
+    end
+
     it "allows modifications of localized values" do
       new_title = 'New Title'
-      version.name['English'] = new_title
-      lang = version.languages.find { |a| a['language'] == 'English' }
+      version.name["English"] = new_title
+      lang = version.languages.find { |a| a['language'] == "English" }
```

Comment:
Prefer `detect` over `find`.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.